frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -373,7 +373,7 @@ function QueryBuilder(props) {
   const { isRunning } = uiControls;
 
   const onTimeout = useCallback(() => {
-    if (Notification.permission === "default") {
+    if ("Notification" in window && Notification.permission === "default") {
```

Comment:
Could we move all `Notification` checks to a module, e.g. to `lib/notifications.ts`? This would prevent copy-pasting these checks across different files.

Reply:
I'd actually vote for hiding all that in the `useWebNotification`. Let's let the fix in and then polish the implementation 🙂